### ISSUE TITLE
Laptop Price Predictor

### ISSUE THREAD

**Issue by yashhibare7** [abhisheks008/DL-Simplified#334]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project title** :Laptop Price Predictor </br>
:red_circle: **Dataset** :kaggle </br>
:red_circle: **Approach** : 
To predict laptop prices using a machine learning approach: 1. Collect a dataset with laptop features and corresponding prices. 2. Preprocess the data by handling missing values, encoding categorical variables, and scaling numerical features. 3. Train a regression model like linear regression, random forest, or gradient boosting to predict laptop prices based on the features. 4. Evaluate the model's performance using metrics like mean squared error or R-squared.
**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : Yash Hibare
- GitHub Profile Link : https://github.com/yashhibare7
- Email ID :yashhibare7@gmail.com
- Participant ID (if applicable):
- Approach for this Project :
- What is your participant role? (Mention the Open Source program)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by yashhibare7** [comment]
Details mentioned

**Comment by abhisheks008** [comment]
This is a deep learning project repository, we expect contributors will come up deep learning methods to solve the problem statements. Please modify your approach and come up again with the new approach by including deep learning methods in it.
@yashhibare7

**Comment by Im-Shivaprakash** [comment]
I would like to work on this project 
Will you assign it to me ?

**Comment by abhisheks008** [comment]
@Im-Shivaprakash what are the deep learning models you are going to implement here?

**Comment by Im-Shivaprakash** [comment]
Sorry i couldnt reply earlier I am going to use feedforwrd network for this project since it is an regression problem

**Comment by Im-Shivaprakash** [comment]
Full name : Shivaprakash S
GitHub Profile Link : https://github.com/Im-Shivaprakash
Email ID :shivaprakashsrinivasaniam@gmail.com
Participant ID (if applicable):
Approach for this Project :I'll be using multilayer perceptron for this project since it is a linear regression problem 
What is your participant role? I dont understand this question, Im new to open source project platform

**Comment by namita0210** [comment]
"I would like to work on this issue. Can you please assign it to me?"

**Comment by namita0210** [comment]
"I would like to work on this issue. Can you please assign it to me?"

**Comment by yashhibare7** [comment]
Okay

**Comment by namita0210** [comment]
can you assign the issue to me or are there are any specific steps I should follow?

**Comment by abhisheks008** [comment]
Hi @namita0210 thanks for showing interest in this project repository. Can you please share your approach, what are the models you are going to use here and how you will solve this issue?

Looking forward to hear from you.

**Comment by namita0210** [comment]
> Hi @namita0210 thanks for showing interest in this project repository. Can you please share your approach, what are the models you are going to use here and how you will solve this issue?
> 
> Looking forward to hear 

I plan to use deep learning models for regression, such as feedforward neural networks, convolutional neural networks (CNNs), or recurrent neural networks (RNNs) to predict laptop prices based on features like CPU, RAM, storage, brand, etc followed by ensemble learning to have improved accuracies.

**Comment by abhisheks008** [comment]
Great approach. In which issue you wanna start working first?
@namita0210

**Comment by YashSachan2** [comment]
Hi @abhisheks008 .Please assign this issue to me.I have worked on a similar problem predicting house prices before.

**Comment by abhisheks008** [comment]
Hi @YashSachan2 can you please share your approach for solving this issue?

**Comment by Dhruv127** [comment]
Can you kindly assign this to me?

**Comment by abhisheks008** [comment]
Hi @Dhruv127 can you please share your approach for solving this issue?

**Comment by onkar-1432** [comment]
Full name : Onkar Viralekar
GitHub Profile Link : https://github.com/onkar-1432
Email ID :onkarviralekar.143@gmail.com
Participant ID (if applicable):
Approach for this Project :I'll be using https://www.kaggle.com/datasets/muhammetvarl/laptop-price these dataset and firstly do some EDA then will go for data cleaning and preprocessing. As these is regression problem will go with different algorithms like linear, random forest, gradient boosting regression first. If the results are satisfactory then will compare the results by using R-squared metric. Else will go for advance method like KNN regression and XGB with optuna for efficient  hyper parameter optimization. Will update you if I find something new in it.
What is your participant role? Codepeak23 participant

**Comment by abhisheks008** [comment]
Try to use CNN/ANN in it. Issue assigned to you @onkar-1432

**Comment by onkar-1432** [comment]
@abhisheks008 i want to continue this issue for SWOC24. Please allow me to continue. I am just about to complete these issue.

**Comment by abhisheks008** [comment]
You are also part of the SWOC event? 😄

**Comment by onkar-1432** [comment]
Yes exploring and participating in all open source events 😄

**Comment by abhisheks008** [comment]
Issue assigned to you @onkar-1432

**Comment by onkar-1432** [comment]
@abhisheks008 created PR for these issue.

**Comment by Gaurav-576** [comment]
I would like to work on this issue and contribute to the repository, please assign this Laptop price predictor to me.

**Comment by abhisheks008** [comment]
Hi @Gaurav-576 nice to have you here. Please follow the issue tenmplate and share all the details here in the comments properly. Otherwise it's difficult to track the contributors.

**Comment by Gaurav-576** [comment]
Sorry about that. Here are my details -
Full name : Gaurav Kumar Singh
GitHub Profile Link : https://github.com/Gaurav-576
Email ID : gauravsingh96753@gmail.com
Participant ID (if applicable): NA
Approach for this Project : First, I will be collecting data and perform some EDA to understand my dataset. I will perform data preprocessing to get rid of any missing or null values and make sure that the labels are marked properly. Extensively, data collection and preprocessing will be the part which will require most time in this project. Then I would be developing a laptop price predictor using an Artificial Neural Network (ANN) with Rectified Linear Unit (ReLU) activation function and Stochastic Gradient Descent (SGD) as the optimizer. I would focus highly on the accuracy of the model and make sure the accuracy is above 90% for both training and test dataset.
What is your participant role? GSSoC'24

**Comment by abhisheks008** [comment]
Hi @Gaurav-576 thanks for sharing your approach. Issue is assigned to you. You can start working on it.

**Comment by Gaurav-576** [comment]
Ok sir, thanks a lot and I will do my best.
